src/exporter/DaemonMetricCollector.cc

```diff
@@ -31,7 +31,17 @@ using json_array = boost::json::array;
 
 void DaemonMetricCollector::request_loop(boost::asio::steady_timer &timer) {
   timer.async_wait([&](const boost::system::error_code &e) {
-    std::cerr << e << std::endl;
+    if (shutdown_flag) {
+      timer.cancel();
+      dout(1) << "Metric collector request loop cancelled" << dendl;
+      return;
+    }
+
+    if (e) {
+      derr << __func__ << ":Error occurred. " << e.message() << dendl;
```

Comment:
if some error has occurred shouldn't we just return from here?

Reply:
if an error occurred during the asynchronous wait  does not mean than the error is going also to be present in next tries/loops ... so better to continue trying to get the metrics.

Before introducing this error management it was only reported the error number (independently if it was an error or not):

It was reported in the log file something like:
```
...
system:0
system:0
...
```

Now we have:
```
Getting metrics loop...
Getting metrics loop...
```

And if an error happens during metrics collection, we will have:

```
Getting metrics loop...
request_loop:Error occurred. blah blah ..
Getting metrics loop...
```

So now, i think that  error management and log output has been improved.